### PR TITLE
canonical sorting

### PR DESCRIPTION
**Pull request by mufernando**
Currently, sort does not put a Table Collection in canonical form (given the order of nodes) -- see #705 .

Here I am working on adding mutation time, mutation parent and node to the mutation sorting key.

(still very early, working on a python impl of sort first!)

### CODE REVIEW

File: python/tests/tsutil.py [hunk 1]
```diff
@@ -680,6 +680,111 @@ def compute_mutation_times(ts):
...
+        randomised_mutations.extend(s.mutations)
+    rng.shuffle(randomised_mutations)
+    for m in randomised_mutations:
+        tables.mutations.add_row(
+            site=site_id_map[m.site],
+            node=m.node,
+            derived_state=m.derived_state,
+            parent=m.parent,
+            metadata=m.metadata,
+            # time=m.time,
```
Comment (jeromekelleher) [inline python/tests/tsutil.py:718]: Time should be working, right?

**Comment by jeromekelleher** [comment]
Looks great @mufernando, this is exatctly the right approach. Ping me when you'd like some feedback.

**Comment by jeromekelleher** [comment]
@mufernando - is this something that we want to get in for the C API release, or will having it done for the next Python release be enough? We'd really like to get the C API shipped soon, and I think just getting the sorting requirement done will be enough for that. Is that OK, or do you want the ``canonicalise`` (or whatever) method to do into the C API in time for the next release too? (The only practical reason for this is if you want to use it in SLiM, I think)

**Comment by mufernando** [comment]
I don't need this in SLiM and probably won't be able to get to this over the next week or two, so we can wait for the next release.

**Comment by benjeffery** [comment]
@mufernando Just pinging as this PR has been inactive for a while - do you plan to pick it up?

**Comment by mufernando** [comment]
I don't think I'll have the time to work on this for a while. But it seems the work I did with validating the C sort with a Python implementation could be incorporated. Is it bad to leave this hanging here?

**Comment by benjeffery** [comment]
It's no problem to leave it - just useful for planning releases to know which PRs are still active.

**Comment by benjeffery** [comment]
Superseded by #1108

**Pull request closed without merging**